### PR TITLE
add try/catch around fs.unlinkSync in Filter.prototype.build to protect against ENOENT, no such file or directory

### PR DESCRIPTION
**Pull request by jcope2013**
related to 

https://github.com/stefanpenner/broccoli-persistent-filter/issues/29
https://github.com/ember-cli/ember-cli/issues/4782#issuecomment-136495069

### CODE REVIEW

**Comment by jcope2013** [comment]
@stefanpenner what should I do with the error? retry the unlink, swallow it, log it?

**Comment by stefanpenner** [comment]
> @stefanpenner what should I do with the error? retry the unlink, swallow it, log it or throw an error?
>  Show all checks

if `error.code === 'ENOENT'` just swallow, otherwise rethrow.

**Comment by jcope2013** [comment]
good to go

File: index.js [hunk 1]
```diff
@@ -70,7 +70,14 @@ Filter.prototype.build = function build() {
   var paths = walkSync(srcDir);
 
   this._cache.deleteExcept(paths).forEach(function(key) {
-    fs.unlinkSync(this.cachePath + '/' + key);
+    try {
+      fs.unlinkSync(this.cachePath + '/' + key);
+    } catch (e) {
+      if(e.code !== 'ENOENT')
+      {
```
Comment (rwjblue) [inline index.js]: Could you move the curly brace up to the same line as the `if` (to follow with the pattern in the rest of the codebase)?  Sorry to be pedantic here.

Comment (stefanpenner) [inline index.js]: :+1:

Comment (jcope2013) [inline index.js]: done

**Pull request merged**